unit-tests/src/org/commcare/android/tests/activities/PostRequestActivityTest.java

```diff
@@ -0,0 +1,205 @@
+
+import android.content.Intent;
+import android.support.v4.util.Pair;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+
+import junit.framework.Assert;
+
+import org.commcare.CommCareApplication;
+import org.commcare.CommCareTestApplication;
+import org.commcare.activities.PostRequestActivity;
+import org.commcare.activities.QueryRequestActivity;
+import org.commcare.android.CommCareTestRunner;
+import org.commcare.android.mocks.HttpURLConnectionMock;
+import org.commcare.android.mocks.ModernHttpRequesterMock;
+import org.commcare.android.util.ActivityLaunchUtils;
+import org.commcare.android.util.TestAppInstaller;
+import org.commcare.dalvik.BuildConfig;
+import org.commcare.dalvik.R;
+import org.commcare.models.AndroidSessionWrapper;
+import org.commcare.network.HttpRequestEndpointsMock;
+import org.commcare.network.LocalDataPullResponseFactory;
+import org.commcare.session.CommCareSession;
+import org.javarosa.core.model.instance.ExternalDataInstance;
+import org.javarosa.core.services.locale.Localization;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.Robolectric;
+import org.robolectric.annotation.Config;
+import org.robolectric.shadows.ShadowActivity;
+
+import java.io.InputStream;
+import java.util.HashMap;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+@Config(application = CommCareTestApplication.class,
+        constants = BuildConfig.class)
+@RunWith(CommCareTestRunner.class)
+public class PostRequestActivityTest {
+    @Before
+    public void setup() {
+        TestAppInstaller.installAppAndLogin(
+                "jr://resource/commcare-apps/case_search_and_claim/profile.ccpr",
+                "test", "123");
+    }
+
+    @Test
+    public void postingToNonHttpsURLTest() {
+        // NOTE PLM: we will eventually support 'http' urls, but won't include authentication credentials in them
+        String badURL = "http://bad.url.com";
+
+        PostRequestActivity postRequestActivity = buildPostActivity(badURL);
+
+        assertErrorMessage(postRequestActivity, true, Localization.get("post.not.using.https", badURL));
+    }
+
+    private static void assertErrorMessage(PostRequestActivity postRequestActivity,
+                                           boolean isVisible,
+                                           String hasMessage) {
+        TextView errorMessage =
+                (TextView)postRequestActivity.findViewById(R.id.error_message);
+        if (isVisible) {
+            Assert.assertEquals(View.VISIBLE, errorMessage.getVisibility());
+        } else {
+            assertFalse(View.VISIBLE == errorMessage.getVisibility());
+        }
+        if (hasMessage != null) {
+            Assert.assertEquals(hasMessage, errorMessage.getText());
```

Comment:
it looks like it doesn't make a difference which you use, but it would be good to be consistent about using the `Assert.` prefix for calls to `assertEquals`, `assertFalse`, etc.